src/test/java/io/openliberty/tools/intellij/it/SingleModProjectTestCommon.java

```diff
@@ -182,277 +207,263 @@ public void testRunTestsActionUsingDropDownMenu() {
+            } finally {
+                // Remove the project from the Liberty tool window.
+                UIBotTestUtils.runActionFromSearchEverywherePanel(remoteRobot, "Liberty: Remove project from the tool window");
 
-        // Start dev mode.
-        UIBotTestUtils.runLibertyTWActionFromDropDownView(remoteRobot, "Start", true);
+                // Select project from the 'Remote Liberty project' dialog.
+                UIBotTestUtils.selectProjectFromRemoveLibertyProjectDialog(remoteRobot, smNLTRestProjectName);
 
-        // Validate that the project started.
-        TestUtils.validateProjectStarted(testName, getSmMpProjResURI(), getSmMpProjPort(), smMPProjOutput, absoluteWLPPath, false);
+                // Answer the 'Remove Liberty project' dialog in the affirmative.
+                UIBotTestUtils.respondToRemoveProjectQueryDialog(remoteRobot);
 
-        try {
-            // Run the project's tests.
-            UIBotTestUtils.runLibertyTWActionFromDropDownView(remoteRobot, "Run tests", true);
+                // Validate that the project was removed. An exception is expected.
+                try {
+                    for (int i = 0; i < 5; i++) {
+                        UIBotTestUtils.findProjectInLibertyToolWindow(remoteRobot, smNLTRestProjectName, "2");
+                        try {
+                            Thread.sleep(2000);
+                        } catch (Exception ee) {
+                            // Nothing to do. Continue to the next iteration.
+                        }
+                    }
 
-            // Validate that the report was generated.
-            validateTestReportsExist();
+                    fail("Project " + smNLTRestProjectName + " is still present in the Liberty tool window despite it being removed.");
+                } catch (Exception e) {
+                    // The project was not found. Success.
+                }
+            }
         } finally {
-            if (TestUtils.isServerStopNeeded(absoluteWLPPath)) {
-                // Stop dev mode.
-                UIBotTestUtils.runLibertyTWActionFromDropDownView(remoteRobot, "Stop", true);
+            // Import the original project.
+            Exception error = null;
+            int maxRetries = 3;
+            for (int i = 0; i < maxRetries; i++) {
+                error = null;
+                try {
+                    UIBotTestUtils.importProject(remoteRobot, projectsPath, smMPProjectName);
+                    UIBotTestUtils.openLibertyToolWindow(remoteRobot);
+                    UIBotTestUtils.expandLibertyToolWindowProjectTree(remoteRobot, smMPProjectName);
+                    break;
+                } catch (Exception e) {
+                    // There are instances in which the import actions are successful, but the
+                    // project view never comes up. Instead, the welcome view is shown. If that is the
+                    // case retry.
```

Comment:
do we need to wait before we retry in this scenario?

Reply:
No not really. The lines that could potentially fail have an inherent wait of a number of seconds before they fail. We could add more wait time if we need to, but so far it does not look like we need it.